diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LambdaLengthRule.kt

```diff
@@ -0,0 +1,76 @@
+package org.cqfn.diktat.ruleset.rules
+
+import org.cqfn.diktat.common.config.rules.RuleConfiguration
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.common.config.rules.getRuleConfig
+import org.cqfn.diktat.ruleset.constants.EmitType
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.utils.*
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.isPartOfComment
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+/**
+ * Rule 5.2.4 check lambda length without parameters
+ */
+class LambdaLengthRule(private val configRules: List<RulesConfig>) : Rule("lambda-length") {
+    private var isFixMode: Boolean = false
+    private lateinit var emitWarn: EmitType
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: EmitType
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val configuration by lazy {
+            LambdaLengthConfiguration(
+                configRules.getRuleConfig(Warnings.TOO_MANY_LINES_IN_LAMBDA)?.configuration ?: emptyMap()
+            )
+        }
+
+        if (node.elementType == ElementType.LAMBDA_EXPRESSION) {
+            checkLambda(node, configuration)
+        }
+    }
+
+    private fun checkLambda(node: ASTNode, configuration: LambdaLengthConfiguration) {
+        val copyNode = node.clone() as ASTNode
+        val sizeLambda = countCodeLines(copyNode)
+        if (sizeLambda > configuration.maxLambdaLength) {
+            val lambdaNodeList = copyNode.findAllNodesWithCondition({it.elementType == ElementType.LAMBDA_EXPRESSION})
+            if (lambdaNodeList.size > 1) {
```

Comment:
Why `> 1`? Is the node itself included?

Reply:
Yes, ignore the node itself

Reply:
`findAllNodesWithCondition` has optional argument `withSelf` to exclude the node, is it suitable here?